GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -902,5 +927,383 @@ public void SwitchFocus(bool alreadyContainedFocus)
+                        GetShortcutKeyDisplayString(Command.StageSelectedFile));
+                }
+
+                DiffText.RevertSelectedContextMenuEntry_Update(
+                    Strings.ResetSelectedLines,
+                    Properties.Images.ResetWorkingDirChanges,
+                    GetShortcutKeyDisplayString(Command.ResetSelectedFiles));
+
+                if (item.IsNew && !FileHelper.IsImage(item.Name))
+                {
+                    // Add visibility, not for normal file displays
+                    DiffText.CherryPickContextMenuEntry_Visible();
+                    DiffText.RevertSelectedContextMenuEntry_Visible();
+                }
+
+                return;
+            }
+
+            DiffText.CherryPickContextMenuEntry_Update(
+                Strings.CherrypickSelectedLines,
+                Properties.Images.CherryPick,
+                GetShortcutKeyDisplayString(Command.StageSelectedFile));
+
+            DiffText.RevertSelectedContextMenuEntry_Update(
+                Strings.RevertSelectedLines,
+                Properties.Images.ResetFileTo,
+                GetShortcutKeyDisplayString(Command.ResetSelectedFiles));
+        }
+
+        /// <summary>
+        /// Check if the commits for the currenly selected item is the same as in FormCommit,
+        /// i.e. HEAD->Index or Index->WorkTree
+        /// </summary>
+        /// <returns>If 'artificial' handling</returns>
+        private bool IsWorkTreeOrIndexWithParent()
+        {
+            if (DiffFiles.SelectedItem == null ||
+                DiffFiles.Revision?.ObjectId == null ||
+                !DiffFiles.Revision.IsArtificial)
+            {
+                return false;
+            }
+
+            var selectedRev = DiffFiles.Revision;
+            var firstId = DiffFiles.SelectedItemParent?.ObjectId;
+
+            return (selectedRev.ObjectId == ObjectId.WorkTreeId
+                    || selectedRev.ObjectId == ObjectId.IndexId)
+                   && selectedRev.FirstParentGuid == firstId;
+        }
+
+        private void StageSelectedLinesToolStripMenuItemClick(object sender, EventArgs e)
+        {
+            StageOrCherryPickSelectedLines();
+        }
+
+        private void StageOrCherryPickSelectedLines()
+        {
+            // to prevent multiple clicks
+            if (!_selectedDiffReloaded)
+            {
+                return;
+            }
+
+            // File no longer selected
+            if (DiffFiles.SelectedItem == null)
+            {
+                return;
+            }
+
+            if (IsWorkTreeOrIndexWithParent())
+            {
+                StageSelectedLines();
+                return;
+            }
+
+            applySelectedLines(false);
+        }
+
+        /// <summary>
+        /// apply patches
+        /// Similar to FileViewer.applySelectedLines()
+        /// </summary>
+        /// <param name="reverse">set if patches is to be reversed</param>
```

Comment:
```suggestion
        /// <summary>
        /// Apply patches.
        /// Similar to FileViewer.ApplySelectedLines()
        /// </summary>
        /// <param name="reverse"><see langword="true"/> if patches is to be reversed; otherwise <see langword="false"/>.</param>
```